--- gdk-pixbuf/gdk-pixbuf-util.c.0	2016-12-20 08:43:26.000000000 -0500
+++ gdk-pixbuf/gdk-pixbuf-util.c	2018-05-30 22:57:54.900626100 -0400
@@ -363,9 +363,13 @@
     {
       gchar *retval;
       /* The localedir is passed to bindtextdomain() which isn't
-      * UTF-8-aware.
-      */
+       * UTF-8-aware.  Try to convert the path back to the legacy
+       * 8-bit locale.  If that's impossible, there's not much we
+       * can do so just return the original build-time path.
+       */
       retval = g_win32_locale_filename_from_utf8 (temp);
+      if (retval == NULL)
+        retval = g_strdup (GDK_PIXBUF_LOCALEDIR);
       g_free (temp);
       return retval;
     }
