--- gcontenttype.c.0	2012-07-14 16:33:18 -0400
+++ gcontenttype.c	2012-08-04 16:02:09 -0400
@@ -332,7 +332,7 @@
 
   /* our test suite and potentially other code used -1 in the past, which is
    * not documented and not allowed; guard against that */
-  g_return_val_if_fail (data_size != (gsize) -1, g_strdup (XDG_MIME_TYPE_UNKNOWN));
+  g_return_val_if_fail (data_size != (gsize) -1, g_strdup ("*"));
 
   if (filename)
     {
