--- gtkrc.0	2009-07-31 06:31:16 -0400
+++ gtkrc	2011-04-02 03:05:34 -0500
@@ -1,4 +1,4 @@
-gtk-icon-sizes = "gtk-menu=13,13:gtk-small-toolbar=16,16:gtk-large-toolbar=24,24:gtk-dnd=32,32"
+gtk-icon-sizes = "gtk-menu=16,16:gtk-button=16,16:gtk-small-toolbar=16,16:gtk-large-toolbar=24,24:gtk-dnd=32,32"
 gtk-toolbar-icon-size = small-toolbar
 
 # disable images in buttons. i've only seen ugly delphi apps use this feature.
