--- gwin32.c.0	2012-04-01 15:34:07 -0400
+++ gwin32.c	2012-04-01 15:33:21 -0400
@@ -396,6 +396,8 @@
 
 #if !defined (_WIN64)
 
+#undef g_win32_get_package_installation_directory
+
 /* DLL ABI binary compatibility version that uses system codepage file names */
 
 gchar *
@@ -473,6 +475,8 @@
 
 #if !defined (_WIN64)
 
+#undef g_win32_get_package_installation_subdirectory
+
 /* DLL ABI binary compatibility version that uses system codepage file names */
 
 gchar *
