--- gwin32.h.0	2013-05-07 14:26:07 -0400
+++ gwin32.h	2013-09-11 19:32:14 -0400
@@ -115,10 +115,8 @@
 #endif	 /* G_PLATFORM_WIN32 */
 
 #ifdef G_OS_WIN32
-#ifdef _WIN64
 #define g_win32_get_package_installation_directory g_win32_get_package_installation_directory_utf8
 #define g_win32_get_package_installation_subdirectory g_win32_get_package_installation_subdirectory_utf8
-#endif
 
 GLIB_AVAILABLE_IN_ALL
 gchar *g_win32_get_package_installation_directory_utf8    (const gchar *package,
