--- iso9660.c.0	2011-04-01 20:17:36 -0500
+++ iso9660.c	2011-04-01 20:16:43 -0500
@@ -811,9 +811,12 @@
 {
   mode_t mode = 0;
 
+#ifdef HAVE_ROCK
   if (yep == p_iso_dirent->rr.b3_rock) {
     return iso9660_get_posix_filemode_from_rock(&p_iso_dirent->rr);
-  } else if (p_iso_dirent->b_xa) {
+  } else 
+#endif
+  if (p_iso_dirent->b_xa) {
     return iso9660_get_posix_filemode_from_xa(p_iso_dirent->xa.attributes);
   }
   return mode;
