--- queryloaders.c.0	2011-12-18 00:51:29 -0500
+++ queryloaders.c	2012-04-02 15:46:26 -0400
@@ -163,7 +163,7 @@
         else
                 path = g_build_filename (dir, file, NULL);
 
-        module = g_module_open (path, 0);
+        module = g_module_open (path, G_MODULE_BIND_LOCAL);
         if (module &&
             g_module_symbol (module, "fill_info", &fill_info_ptr) &&
             g_module_symbol (module, "fill_vtable", &fill_vtable_ptr)) {
@@ -207,8 +207,8 @@
                 else
                         g_fprintf (stderr, "Cannot load loader %s\n", path);
         }
-        if (module)
-                g_module_close (module);
+//        if (module)
+//                g_module_close (module);
         g_free (path);
 }
 
